### PR TITLE
Persistence

### PR DESCRIPTION
**Pull request by mkalinin**
### What was done?
- A new database kind added; it featured with write buffer and designed to work with various storage engines like RocksDB
- Two flushing strategies is implemented:
  - instant: emits flushing on each commit
  - buffer size observer: emits flushing each time when buffer size exceeds predefined limit
- RocksDB source is implemented
- Basic tests for database, rocksdb source and their superposition
- database of `node` executable is switched to RocksDB

### CODE REVIEW

**Review by Nashatyrev — APPROVED** [review]
Looks nice

**Review by zilm13 — CHANGES_REQUESTED** [review]
LGTM
but tests are failed

File: start/common/src/main/java/org/ethereum/beacon/start/common/NodeLauncher.java [hunk 1]
```diff
@@ -123,7 +124,7 @@ void chainStarted(ChainStart chainStartEvent) {
         new ExtendedSlotTransition(perEpochTransition, perSlotTransition, spec);
     emptySlotTransition = new EmptySlotTransition(extendedSlotTransition);
 
-    db = new InMemoryDatabase();
+    db = Database.rocksDB(Paths.get(computeDbName(chainStartEvent)).toString(), 64L << 20);
```
Comment (zilm13) [inline start/common/src/main/java/org/ethereum/beacon/start/common/NodeLauncher.java]: what does this magic number means?

**Review by zilm13 — APPROVED** [review]
LGTM

File: start/common/src/main/java/org/ethereum/beacon/start/common/NodeLauncher.java (hunk 1 above)
Comment (mkalinin) [reply start/common/src/main/java/org/ethereum/beacon/start/common/NodeLauncher.java]: Fixed!

**Pull request merged**